aea/helpers/acn/agent_record.py

```diff
@@ -0,0 +1,115 @@
+# -*- coding: utf-8 -*-
+# ------------------------------------------------------------------------------
+#
+#   Copyright 2018-2019 Fetch.AI Limited
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+# ------------------------------------------------------------------------------
+
+"""This module contains types and helpers for acn Proof-of-Representation."""
+
+import base64
+from hashlib import sha256
+
+from ecdsa import SECP256k1, VerifyingKey
+from ecdsa import util as ecdsa_util
+
+from aea.crypto.fetchai import FetchAIHelper
+
+
+def verify_message(msg: bytes, signature: str, public_key: str) -> bool:
+    """
+    Verifies signature generate by Crypto.sign_message()
+
+    :param msg: original bytes
+    :param signature: signature previousely generated by sign_message()
+    :param public_key: verifying key
+    :return: True if signature is valid
+    """
+    key = VerifyingKey.from_string(
+        bytes.fromhex(public_key), curve=SECP256k1, hashfunc=sha256
+    )
+    signature_bytes = base64.b64decode(signature)
+    return key.verify(
+        signature_bytes, msg, hashfunc=sha256, sigdecode=ecdsa_util.sigdecode_string
+    )
+
+
+class AgentRecord:
+    """Agent Proof-of-Representation to peer"""
+
+    def __init__(
+        self,
+        address: str,
+        public_key: str,
+        peer_public_key: str,
+        signature: str,
+        service_id: str,
+    ):
+        """Initialize the AgentRecord"""
+        self._service_id = service_id
+        self._address = address
+        self._public_key = public_key
+        self._peer_public_key = peer_public_key
+        self._signature = signature
```

Comment:
describe arguments either in class docstring or in __init__ docstring

Reply:
also, shouldn't there be also `valid_until` as we discussed?
Actually, we should have `not_before` and `not_after`, as used by other standards.